### DISCUSSION TITLE
does  plugins in folder bootstrap-plugins support forkjoinpool

### DISCUSSION THREAD

**Discussion by peterpanzsy** [apache/skywalking#9705] [Q&A] [answered]
I have put plugins in bootstrap-plugins to plugins, but calls in forkjoinpool-tasks can't be traced
https://skywalking.apache.org/docs/skywalking-java/v8.10.0/en/setup/service-agent/java-agent/bootstrap-plugins/

**Comment by wu-sheng** [comment]
Why do you think you can? The docs never mention this.

**Reply by peterpanzsy** [reply]
So it doesn't support forjoinpool, then i tried runnable
I put three bootstrap plugins into plugins folder,  restart my application, but the log printed by thread lost the traceId

followed this: https://skywalking.apache.org/docs/skywalking-java/v8.10.0/en/setup/service-agent/java-agent/bootstrap-plugins/#bootstrap-class-plugins
![image](https://user-images.githubusercontent.com/5291999/192928568-3f78f8fe-df2d-42c9-9992-b71434a427fd.png)
my application trace log:
![image](https://user-images.githubusercontent.com/5291999/192928782-58c62264-e86e-4958-ae40-aa20da4b84bd.png)

**Reply by peterpanzsy** [reply]
And I also found trace-cross-thread SDK:    https://skywalking.apache.org/docs/skywalking-java/v8.10.0/en/setup/service-agent/java-agent/application-toolkit-trace-cross-thread/#trace-cross-thread

I wonder what's the relationship bewteen thread SDK and thread-support plugins?

**Reply by zuisong** [reply]
is there any plan to support `forkjoinpool` ❓  😃

**Comment by wu-sheng** [comment, marked as answer]
This has been added through https://github.com/apache/skywalking-java/pull/505

**Reply by yiwenup** [reply]
May I ask how can i enhance java.lang.reflect.Mehtod#invoke like bootstrap-plugins?

**Reply by wu-sheng** [reply]
I don't think you should do that. It should be loaded in JVM, more importantly, it will impact your whole JVM performance.
